lib/view_component/compiler.rb

```diff
@@ -121,13 +126,23 @@ def template_errors
 
     def templates
       @templates ||= begin
-        extensions = ActionView::Template.template_handler_extensions
+        extensions = %w[builder erb haml html jbuilder raw ruby slim]
 
         component_class._sidecar_files(extensions).each_with_object([]) do |path, memo|
           pieces = File.basename(path).split(".")
+          variant = pieces.second.split("+").second&.to_sym
+
+          # grab localized templates and fix regional dash (pt-BR, en-GB...)
+          localized_template = pieces.second.downcase.tr("-", "_")
+
+          # if template is a localized (ie file.country_code.html.erb) append as variant
+          if !localized_template.match(/builder|erb|html|jbuilder|raw|ruby|slim|tablet|iphone|json|html|\+/).present?
```

Comment:
Same here, can we avoid hardcoding (or at least reuse the values from above)?

Reply:
I changed how identify it, take a look in the next commit.